Source/WebCore/animation/ViewTimeline.cpp

```diff
@@ -140,4 +142,67 @@ AnimationTimeline::ShouldUpdateAnimationsAndSendEvents ViewTimeline::documentWil
     return AnimationTimeline::ShouldUpdateAnimationsAndSendEvents::No;
 }
 
+Element* ViewTimeline::source() const
+{
+    if (auto* sourceRender = sourceRenderer())
```

Comment:
I would suggest flipping this around; keep a reference to the element, and derive the renderer from it. Renderers can change (`display:none` toggled etc).

Reply:
We aren't holding a reference here just getting the result from `enclosingScrollableContainer`.